docs/learning/frontend/Localization.md

```diff
@@ -31,42 +33,129 @@ For example, suppose you are developing an application called SafetyBase and you
 }
 ```
 
-The messages can now be accessed by first registering the namespace, and then using the translate method:
+`public/locales/de/SafetyBaseMessages.json`
 
-```ts
-messageNS: I18NNamespace = IModelApp.registerNamespace ("SafetyBaseMessages");
-await messageNS.readFinished;
-if (this.notLoggedIn) {
-  console.log (IModelApp.i18n.translate("SafetyBaseMessages:info.login.loggedIn")
-} else {
-  console.log (IModelApp.i18n.translate("SafetyBaseMessages:info.login.notLoggedIn", {userName: this.loginName});
+```json
+{
+  "info": {
+    "login": {
+      "notLoggedIn": "Sie sind derzeit nicht eingeloggt.",
+      "loggedIn": "Sie sind angemeldet als {{username}}."
+    }
+  },
+  "warning": {
+    "login": {
+      "mustLogin": "Diese Funktion ist nur verfügbar, wenn Sie sich anmelden.",
+      "notAuthorized": "Sie sind nicht berechtigt, auf diese Ressource zuzugreifen."
+    }
+  },
+  "error": {
+    "loginIncorrect": "Die Kombination aus Benutzername und Passwort ist ungültig.",
+    "offline": "NNetzwerkverbindung nicht verfügbar."
+  }
 }
 ```
 
-In the example above, we start by registering the namespace with IModelApp. That starts the process of retrieving the SafetyBaseMessages.json file in the directory corresponding to the current locale (in this case, "en") from the server. Since that might take a little while, before the first use of a namespace, we await on the readFinished property of the I18NNamespace, which is a Promise that is fulfilled when the file is retrieved and ready to be accessed by the translate method. If not logged in, we use the simple form of the translate method to display the string "You are not currently logged in to Bentley Connect." to the console. If the user is logged in, the message "You are logged in as xxx.", is displayed on the console, where xxx is replaced by this.loginName. This demonstrates passing additional argument to the translate method, which substitutes the value of the arguments for the corresponding variables specified in the {{ }} formulas in the translation string. That substitution is called "interpolation" in internationalization terminology.
+Note: German was translated from English in Google Translate.
+
+By default, the IModelApp uses an instance of [ITwinLocalization]($i18n) for localization purposes. Below, a user-created instance of this class will be passed in to IModelApp to demonstrate how a custom [Localization]($common) implementation could be used instead.
+
+```ts
+import { IModelApp } from "@itwin/core-frontend";
+
+// The Localization interface class
+import { Localization } from "@itwin/core-common";
+
+// The iTwin.js implementation for Localization.
+// *This can be replaced with your own implementation if you want!
+import { ITwinLocalization } from "@itwin/core-i18n";
+
+var myCustomLocalization: Localization = new ITwinLocalization();
```

Comment:
Change var to const

Reply:
FYI @mattbjordan if you add your example code to the [example code snippets package](https://github.com/iTwin/itwinjs-core/tree/master/example-code/snippets/src/frontend) it will be guaranteed to compile now and in the future. This is generally a good idea for snippets longer than a couple of lines.

Reply:
Let @mgooding decide whether to resolve comments please.